neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/presenter/EditVariantPresenter.kt

```diff
@@ -40,6 +50,21 @@ class EditVariantPresenter @Inject constructor(): Presenter {
     fun onBackClicked() {
         if(newVariant && variant.name.isNullOrEmpty()) {
             view.goToVariantList(false, "")
+        } else if(originalVariant != null) {
+            // Check if any changes would be lost
+            var changed = false;
+            for(field in variant.configuration!!.javaClass.serializableFields) {
+                if(!field.get(variant.configuration).equals(field.get(originalVariant!!.configuration))) {
+                    changed = true
```

Comment:
you could probably cut down on this flag and just call view.createCancelConfirmation() here instead. It would mean a mid-function return statement though. up to you, but I find flags for iterative checks like this a bit redundant